lib/xlat_tables_v2/xlat_tables_private.h

```diff
@@ -108,6 +108,13 @@ typedef struct {
 	/* Set to 1 when the translation tables are initialized. */
 	int initialized;
 
+	/*
+	 * Bit mask that has to be ORed to the rest of a translation table
+	 * descriptor in order to prohibit execution of code at the exception
+	 * level of this translation context.
+	 */
+	uint64_t execute_never_mask;
```

Comment:
An earlier version did as you suggest but this means that a calculation of the mask to use from the EL needs to be done for every xlat table created. This way is more efficient. If the EL is needed in the context later, this can be easily added without affecting compatibility.